app/commands/warp.js

```diff
@@ -14,70 +14,220 @@ module.exports = Base => class extends Base {
             { text: '!warp ', color: 'green' },
             { text: 'PlayerName ', color: 'aqua' },
             'sends a warp request to another player.\n',
-            'They will have to accept the request first.\n(has 15 minute cooldown if accepted)\n\n',
             { text: '!warp accept', color: 'green' },
-            ' accepts a warp request from another player.',
+            ' accepts a warp request from the next player in the queue.\n',
+            { text: '!warp decline', color: 'green' },
+            ' declines a warp request from the next player in the queue.\n',
+            { text: '!warp queue', color: 'green' },
+            ' lists players who have sent you a warp request.\n',
+            { text: '!warp accept/decline all', color: 'green' },
+            ' accepts or declines warp requests from all players in the queue.',
         ];
     }
 
-    async handleWarp(playerName, args) {
+    async handleWarp(playerName, args) {       
         // Check to see if the player is in the end if so don't allow anything home related
         // I could have checked upon request if the recipient is in the end but decided it wasn't worth the extra code
         if ((await this.getPlayerDimension(playerName)) === 'minecraft:the_end') {
-            return this.tellPlayer(playerName, 'Sorry can not use the warp command in the end', 'red');
+            return this.tellPlayer(playerName, 'Sorry cannot use the warp command in the end', 'red');
         }
+        
+        //choose a function based on the args.   
+        if (!args[0]) {
+            return this.handleWrongWarpInput(playerName);
+        } else {
+            let warpTo;
+            let loggedInPlayers = await this.getListOfOnlinePlayers();
+            if (loggedInPlayers.indexOf(args[0]) !== -1) {
+                // If the first word is a players name then make a request for warp
+                let index = loggedInPlayers.indexOf(args[0]);
```

Comment:
Whay don't you just save `args[0]` as  `warpTo` or `playerName` instead of finding the index just to pull it out?

Reply:
because that's logical josh and im never logical.